apps/portal/modules/tokenUtil.js

```diff
@@ -0,0 +1,375 @@
+     *                      "applicationType": "",
+     *                      "grantType": "",
+     *                      "saasApp" :"",
+     *                      "dynamicClientRegistrationEndPoint" : ""
+     *
+     * @returns {{clientId:*, clientSecret:*}}
+     */
+    module.getDynamicCredentials = function (properties) {
+        var payload = {
+            "callbackUrl": properties.callBackUrl,
+            "clientName": properties.clientName,
+            "tokenScope": properties.tokenScope,
+            "owner": properties.owner,
+            "applicationType": properties.applicationType,
+            "grantType": properties.grantType,
+            "saasApp": properties.saasApp
+        };
+        var xhr = new XMLHttpRequest();
+        var tokenEndpoint = properties.dynamicClientRegistrationEndPoint;
+        xhr.open(constants.HTTP_POST, tokenEndpoint, false);
+        xhr.setRequestHeader(constants.CONTENT_TYPE_IDENTIFIER, constants.APPLICATION_JSON);
+        xhr.send(payload);
+        var clientData = {};
+        if (xhr.status == 201) {
+            var data = parse(xhr.responseText);
+            clientData.clientId = data.client_id;
+            clientData.clientSecret = data.client_secret;
+        } else if (xhr.status == 400) {
+            throw "While creating an OAuth application. Invalid client meta data";
+        } else {
+            throw "Error while creating an OAuth application, obtaining client id and secret";
+        }
+        return clientData;
+    };
+
+    /**
+     * Encode the payload in Base64
+     * @param payload
+     * @returns {Packages.java.lang.String}
+     */
+    module.encode = function (payload) {
+        return new String(Base64.encodeBase64(new String(payload).getBytes()));
+    }
+
+    /**
+     * Decode the payload in Base64
+     * @param payload
+     * @returns {Packages.java.lang.String}
+     */
+    module.decode = function (payload) {
+        return new String(Base64.decodeBase64(new String(payload).getBytes()));
+    }
+
+    /**
+     * Get an AccessToken pair based on username and password
+     * @param username
+     * @param password
+     * @param encodedClientKeys  {{clientId:"", clientSecret:""}}
+     * @param scope              eg: PRODUCTION
+     * @returns {{accessToken: *, refreshToken: *}}
+     */
+    module.getTokenWithPasswordGrantType = function (username, password, encodedClientKeys, scope, idPServer) {
+        var xhr = new XMLHttpRequest();
+        var tokenEndpoint = idPServer;
+        xhr.open(constants.HTTP_POST, tokenEndpoint, false);
+        xhr.setRequestHeader(constants.CONTENT_TYPE_IDENTIFIER, constants.APPLICATION_X_WWW_FOR_URLENCODED);
+        xhr.setRequestHeader(constants.AUTHORIZATION_HEADER, constants.BASIC_PREFIX + encodedClientKeys);
+        xhr.send("grant_type=password&username=" + username + "&password=" + password + "&scope=" + scope);
+        delete password, delete clientSecret, delete encodedClientKeys;
```

Comment:
Delete should only be used to delete properties of an object, not a variable defined (This won't work in strict mode)